### PR TITLE
Feature/use battery part

### PR DESCRIPTION
**Pull request by PetervDooren**
Created for people who occasionally forget that the robot is charging :)

### CODE REVIEW

**Comment by PetervDooren** [comment]
I moved a lot of checks to a new state 'prepare drive' which ensures that the robot is ready to start driving. The one thing that could not be placed here is going to the driving pose, because that might change based on the distance we need to drive.

This does add the assumption that the things in the 'prepare drive' state only need to be checked at the start of navigation and cannot be changed during the process. i.e. noone plugs in the robot while navigating and noone uses and then releases the head.

File: robot_smach_states/src/robot_smach_states/navigation/navigation.py [hunk 1]
```diff
@@ -39,6 +40,45 @@ def execute(self, userdata=None):
...
+
+        # Cancel head goal, we need it for navigation :)
+        if self.reset_head:
+            self.robot.head.close()
+
+        for part in self.robot.parts.itervalues():
+            # ensure all batteries are unplugged
+            if isinstance(part, Battery):
+                while part.is_charging:
+                    if tstart - rospy.Time.now() < rospy.Duration.from_sec(self.timeout):
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/navigation/navigation.py:70]: I am not sure this needs to be and `rospy.duration`. Maybe a float/int would also work

Comment (MatthijsBurgh) [reply robot_smach_states/src/robot_smach_states/navigation/navigation.py:70]: First part is going to b e negative.... I don't think that should be the case. Also I prefer now > start+timeout

File: robot_smach_states/src/robot_smach_states/navigation/navigation.py [hunk 2]
```diff
@@ -39,6 +40,45 @@ def execute(self, userdata=None):
...
+                        return 'abort'
+                    if self.preempt_requested():
+                        return 'preempted'
+                    rospy.logwarn("Robot is charging. Please unplug")
+                    self.robot.speech.speak("I am still charging. Please unplug me.")
+                    rospy.sleep(1)
+                rospy.logerr("Robot is charging, aborting navigation")
+                return 'abort'
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/navigation/navigation.py:78]: This code is reaced when not charging anymore. Am I correct? That is not matching the code.

File: robot_smach_states/src/robot_smach_states/navigation/navigation.py [hunk 3]
```diff
@@ -80,13 +120,12 @@ def execute(self, userdata=None):
 
 
 class executePlan(smach.State):
-    def __init__(self, robot, breakout_function, blocked_timeout=4, reset_head=True, reset_pose=True):
-        smach.State.__init__(self, outcomes=['succeeded', 'arrived', 'blocked', 'preempted'])
+    def __init__(self, robot, breakout_function, blocked_timeout = 4, reset_pose=True):
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/navigation/navigation.py:123]: No space around equal sings in default arguments

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
Also fix CI

File: robot_smach_states/src/robot_smach_states/navigation/navigation.py [hunk 4]
```diff
@@ -80,13 +118,12 @@ def execute(self, userdata=None):
 
 
 class executePlan(smach.State):
-    def __init__(self, robot, breakout_function, blocked_timeout=4, reset_head=True, reset_pose=True):
```
Comment (MatthijsBurgh) [inline robot_smach_states/src/robot_smach_states/navigation/navigation.py:83]: Why dropping this? Is this only obsolete for HERO or all robots?

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
CI fails on missing `reset_head`

**Comment by MatthijsBurgh** [comment]
I think we can delete the branch. GH will still keep the diff.

**Pull request closed without merging**